sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -336,8 +336,14 @@ static int sof_rt5682_hw_params(struct snd_pcm_substream *substream,
 		clk_id = RT5645_SCLK_S_PLL1;
 		break;
 	case CODEC_RT5682:
-		pll_id = RT5682_PLL1;
-		clk_id = RT5682_SCLK_S_PLL1;
+		/* For MCLK = 24.576MHz and sample rate = 96KHz case, use PLL2 */
```

Comment:
I don't see any test of the sample rate, this is selecting PLL2 always for MTL, isn't it?

Reply:
@bardliao I think you have the logic reversed. I think we should do ```if (pll_out == 24576000), use PLL1 else use PLL2```

Reply:
> I don't see any test of the sample rate, this is selecting PLL2 always for MTL, isn't it?

I changed in the wrong case. It should be CODEC_RT5682S instead. But, yes, MLT will always use RT5682S_PLL1 because RT5682S_PLL2 doesn't support 24.576MHz input.

Reply:
> @bardliao I think you have the logic reversed. I think we should do `if (pll_out == 24576000), use PLL2 else use PLL1`

No, test pll_out is not quite right in the rt5682s pll selection case. [plla_table[]](https://github.com/thesofproject/linux/blob/f3f97ec8d35336322e21d6489eb87a3bcd98b323/sound/soc/codecs/rt5682s.c#L2255) is for RT5682S_PLL1 and plla_table[] -> [pllb_table[]](https://github.com/thesofproject/linux/blob/f3f97ec8d35336322e21d6489eb87a3bcd98b323/sound/soc/codecs/rt5682s.c#L2279) is for PLL2. Looks at the 2 tables, we need to use PLL2 when pll_in = 48000000, 24000000, 19200000, or 38400000 when pll_out is 49152000.